wallet/chainntfns_test.go

```diff
@@ -0,0 +1,346 @@
+	return s.birthday
+}
+
+// BirthdayBlock returns the birthday block of the wallet.
+func (s *mockSyncStore) BirthdayBlock(
+	walletdb.ReadBucket) (waddrmgr.BlockStamp, bool, error) {
+
+	if s.birthdayBlock == nil {
+		err := waddrmgr.ManagerError{
+			ErrorCode: waddrmgr.ErrBirthdayBlockNotSet,
+		}
+		return waddrmgr.BlockStamp{}, false, err
+	}
+
+	return *s.birthdayBlock, s.birthdayBlockVerified, nil
+}
+
+// SetBirthdayBlock updates the birthday block of the wallet to the given block.
+// The boolean can be used to signal whether this block should be sanity checked
+// the next time the wallet starts.
+func (s *mockSyncStore) SetBirthdayBlock(_ walletdb.ReadWriteBucket,
+	block waddrmgr.BlockStamp, verified bool) error {
+
+	s.birthdayBlock = &block
+	s.birthdayBlockVerified = verified
+	return nil
+}
+
+// SetSyncedTo marks the address manager to be in sync with the recently-seen
+// block described by the blockstamp. When the provided blockstamp is nil, the
+// oldest blockstamp of the block the manager was created at and of all imported
+// addresses will be used. This effectively allows the manager to be marked as
+// unsynced back to the oldest known point any of the addresses have appeared in
+// the block chain.
+func (s *mockSyncStore) SetSyncedTo(_ walletdb.ReadWriteBucket,
+	block *waddrmgr.BlockStamp) error {
+
+	s.syncedTo = *block
+	return nil
+}
+
+// TestBirthdaySanityCheckEmptyBirthdayBlock ensures that a sanity check is not
+// done if the birthday block does not exist in the first place.
+func TestBirthdaySanityCheckEmptyBirthdayBlock(t *testing.T) {
+	t.Parallel()
+
+	db, cleanup := createTestDB(t)
+	defer cleanup()
+
+	chainConn := &mockChainConn{}
+
+	// Our sync store will reflect that we don't have a birthday block set,
+	// so we should not attempt a sanity check.
+	syncStore := &mockSyncStore{}
+
+	birthdayBlock, err := birthdaySanityCheck(db, chainConn, syncStore)
+	if err != nil {
+		t.Fatalf("unable to sanity check birthday block: %v", err)
+	}
+
+	if birthdayBlock != nil {
+		t.Fatalf("expected birthday block to be nil due to not being "+
+			"set, got %v", *birthdayBlock)
+	}
+}
+
+// TestBirthdaySanityCheckVerifiedBirthdayBlock ensures that a sanity check is
+// not performed if the birthday block has already been verified.
+func TestBirthdaySanityCheckVerifiedBirthdayBlock(t *testing.T) {
+	t.Parallel()
+
+	db, cleanup := createTestDB(t)
+	defer cleanup()
+
+	const chainTip = 5000
+	chainConn := createMockChainConn(chainParams.GenesisBlock, chainTip)
+
+	// Our sync store reflects that our birthday block has already been
+	// verified and should not require a sanity check.
+	syncStore := &mockSyncStore{
+		birthdayBlock: &waddrmgr.BlockStamp{
+			Hash:   chainhash.Hash{},
+			Height: 1337,
+		},
+		birthdayBlockVerified: true,
+		syncedTo: waddrmgr.BlockStamp{
+			Height: chainTip,
+		},
+	}
+
+	if _, err := birthdaySanityCheck(db, chainConn, syncStore); err != nil {
```

Comment:
Should check that the block is returned since we expect to pass it up to the caller.